python/cuml/experimental/common/base.pyx

```diff
@@ -0,0 +1,535 @@
+        Sets logging level. It must be one of `cuml.common.logger.level_*`.
+        See :ref:`verbosity-levels` for more info.
+    output_type : {'input', 'cudf', 'cupy', 'numpy', 'numba'}, default=None
+        Variable to control output type of the results and attributes of
+        the estimator. If None, it'll inherit the output type set at the
+        module level, `cuml.global_settings.output_type`.
+        See :ref:`output-data-type-configuration` for more info.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        from cuml import Base
+
+        # assuming this ML algo has separate 'fit' and 'predict' methods
+        class MyAlgo(Base):
+            def __init__(self, ...):
+                super(MyAlgo, self).__init__(...)
+                # other setup logic
+
+            def fit(self, data, ...):
+                # check output format
+                self._check_output_type(data)
+                # train logic goes here
+
+            def predict(self, data, ...):
+                # check output format
+                self._check_output_type(data)
+                # inference logic goes here
+
+            def get_param_names(self):
+                # return a list of hyperparam names supported by this algo
+
+        # stream and handle example:
+
+        stream = cuml.cuda.Stream()
+        handle = cuml.Handle(stream=stream)
+
+        algo = MyAlgo(handle=handle)
+        algo.fit(...)
+        result = algo.predict(...)
+
+        # final sync of all gpu-work launched inside this object
+        # this is same as `cuml.cuda.Stream.sync()` call, but safer in case
+        # the default stream inside the `cumlHandle` is being used
+        base.handle.sync()
+        del base  # optional!
+    """
+
+    def __init__(self, *,
+                 handle=None,
+                 verbose=False,
+                 output_type=None):
+        """
+        Constructor. All children must call init method of this base class.
+
+        """
+        self.handle = raft.common.handle.Handle() if handle is None \
+            else handle
+
+        # Internally, self.verbose follows the spdlog/c++ standard of
+        # 0 is most logging, and logging decreases from there.
+        # So if the user passes an int value for logging, we convert it.
+        if verbose is True:
+            self.verbose = logger.level_debug
+        elif verbose is False:
+            self.verbose = logger.level_info
+        else:
+            self.verbose = verbose
+
+        self.output_type = _check_output_type_str(
+            cuml.global_settings.output_type
+            if output_type is None else output_type)
+        self._input_type = None
+        self.target_dtype = None
+        self.n_features_in_ = None
+
+        nvtx_benchmark = os.getenv('NVTX_BENCHMARK')
+        if nvtx_benchmark and nvtx_benchmark.lower() == 'true':
+            self.set_nvtx_annotations()
+
+    def __repr__(self):
```

Comment:
We might want to add some more explicit printing for some of the important attributes that will help us debug like whether the estimator was trained on cpu or gpu